### PR TITLE
docs(guides): add more detail to code-signing

### LINKED ISSUE #25763: Improve code signing docs
Recent token rotation revealed some gaps in the [code signing docs](https://github.com/cypress-io/cypress/blob/develop/guides/code-signing.md). Add documentation around the following:

- [ ] Receiving dev program certificate + private key via xcode
- [ ] How to create a CSC_LINK
- [ ] Anything else that caused confusion during rotation

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #25763 

### Additional details

- Adds more specific details around rotating the Mac/Windows code signing credentials.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by chrisbreiding — APPROVED** [review]

File: guides/code-signing.md [hunk 1]
```diff
@@ -4,20 +4,36 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
-## Installing a new Mac code signing key
+## Rotating the Mac code signing key
 
-Follow the directions supplied by `electron-builder`: https://www.electron.build/code-signing#travis-appveyor-and-other-ci-servers
+1. On a Mac, log in to Xcode using Cypress's Apple developer program identity.
+2. Follow Apple's [Create, export, and delete signing certificates](https://help.apple.com/xcode/mac/current/#/dev154b28f09) instructions:
+    1. Follow "View signing certificates".
+    2. Follow "Create a signing certificate", and choose the type of "Developer ID Application" when prompted.
+    3. Follow "Export a signing certificate". Set a strong passphrase when prompted, which will later become `CSC_KEY_PASSWORD`.
+3. Upload the exported, encrypted `.p12` file to the Cypress App Google Drive and obtain a public [direct download link][direct-download].
```
Comment (emilyrohrbough) [inline guides/code-signing.md]: Where does this go in the shared drive?

File: guides/code-signing.md [hunk 2]
```diff
@@ -4,20 +4,36 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
+4. Within the `test-runner:sign-mac-binary` CircleCI context, set `CSC_LINK` to that direct download URL and set `CSC_KEY_PASSWORD` to the passphrase used to encrypt the `pfx` file.
```
Comment (emilyrohrbough) [inline guides/code-signing.md]: `pfx` or `.p12` file?

File: guides/code-signing.md [hunk 3]
```diff
@@ -4,20 +4,36 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
 
-Set the environment variables `CSC_LINK` and `CSC_KEY_PASSWORD` in the `test-runner:sign-mac-binary` CircleCI context.
+## Rotating the Windows code signing key
 
-## Installing a new Windows code signing key
-
-1. Obtain the private key and full certificate chain in ASCII-armored PEM format and store each in a file (`-----BEGIN PRIVATE KEY-----`, `-----BEGIN CERTIFICATE-----`)
-2. Using `openssl`, convert the plaintext PEM public and private key to binary PKCS#12/PFX format and encrypt it with a real strong password.
+1. Generate a certificate signing request (CSR) file using `openssl`. For example:
```
Comment (emilyrohrbough) [inline guides/code-signing.md:19]: is there a specific openssl version needed?

File: guides/code-signing.md [hunk 4]
```diff
@@ -4,20 +4,36 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
+    ```shell
+    # generate a new private key
+    openssl genrsa -out win-code-signing.key 4096
+    # create a CSR using the private key
+    openssl req -new -key win-code-signing.key -out win-code-signing.csr
+    ```
+2. Obtain a certificate by submitting your CSR to SSL.com using the Cypress SSL.com account.
```
Comment (emilyrohrbough) [inline guides/code-signing.md]:
```suggestion
2. Obtain a certificate by submitting the CSR to SSL.com using the Cypress SSL.com account.
```

File: guides/code-signing.md (hunk 3 above)
Comment (flotwig) [reply guides/code-signing.md:19]: AFAIK no, this should work on any modern openssl version. Feel free to validate locally, I checked on Linux.

File: guides/code-signing.md [hunk 5]
```diff
@@ -4,20 +4,36 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
+    * If renewing, follow the [renewal instructions](https://www.ssl.com/how-to/renewing-ev-ov-and-iv-certificates/).
```
Comment (emilyrohrbough) [inline guides/code-signing.md:27]: When would one renew vs rotate? Only renew if it expired? ideally we don't get to this point right? ALso how long are these certs good for? 6 months? Do we update these in the shared password vault as well with the expiration dates?

File: guides/code-signing.md (hunk 2 above)
Comment (flotwig) [reply guides/code-signing.md]: whoops:

```suggestion
4. Within the `test-runner:sign-mac-binary` CircleCI context, set `CSC_LINK` to that direct download URL and set `CSC_KEY_PASSWORD` to the passphrase used to encrypt the `p12` file.
```

File: guides/code-signing.md [hunk 6]
```diff
@@ -4,20 +4,36 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
+    * If rotating, contact SSL.com's support to request certificate re-issuance.
+3. Obtain the full certificate chain from SSL.com's dashboard in ASCII-armored PEM format and save it as `win-code-signing.crt`. (`-----BEGIN PRIVATE KEY-----`, `-----BEGIN CERTIFICATE-----`)
+4. Using `openssl`, convert the plaintext PEM public and private key to binary PKCS#12/PFX format and encrypt it with a strong passphrase, which will later become `CSC_KEY_PASSWORD`.
     ```shell
-    ➜ openssl pkcs12 -export -inkey key.pem -in cert.pem -out encrypted.pfx
+    ➜ openssl pkcs12 -export -inkey win-code-signing.key -in win-code-signing.crt -out encrypted-win-code-signing.pfx
     Enter Export Password: <password>
```
Comment (emilyrohrbough) [inline guides/code-signing.md:33]:
```suggestion
    Enter Export Password: <passphrase>
```

**Review by emilyrohrbough — APPROVED** [review]
Nice job on this! Left a few comments but feel like I'd know how to do this now! 🚀

File: guides/code-signing.md (hunk 1 above)
Comment (flotwig) [reply guides/code-signing.md]: Added direct links: [9296776](https://github.com/cypress-io/cypress/pull/25794/commits/9296776bdd254ed9c6dac77b8791da01613a6f68)

File: guides/code-signing.md (hunk 5 above)
Comment (flotwig) [reply guides/code-signing.md:27]: Rotate if leaked, renew if it's about to expire. Ideally yeah... I can add something to the team calendar to try and act as a reminder. Windows code signing certs last for 3 years max, but we have a 10 year plan with SSL.com so we can renew it a few times without buying again. I'll have to dig deeper into the Apple cert expiry time.

File: guides/code-signing.md [hunk 7]
```diff
@@ -4,20 +4,37 @@ Code signing is done for the Windows and Mac distributions of Cypress when they
...
+2. Obtain a certificate by submitting the CSR to SSL.com using the Cypress SSL.com account.
+    * If renewing, follow the [renewal instructions](https://www.ssl.com/how-to/renewing-ev-ov-and-iv-certificates/).
+    * If rotating, contact SSL.com's support to request certificate re-issuance.
+3. Obtain the full certificate chain from SSL.com's dashboard in ASCII-armored PEM format and save it as `win-code-signing.crt`. (`-----BEGIN PRIVATE KEY-----`, `-----BEGIN CERTIFICATE-----`)
+4. Using `openssl`, convert the plaintext PEM public and private key to binary PKCS#12/PFX format and encrypt it with a strong passphrase, which will later become `CSC_KEY_PASSWORD`.
     ```shell
-    ➜ openssl pkcs12 -export -inkey key.pem -in cert.pem -out encrypted.pfx
-    Enter Export Password: <password>
+    ➜ openssl pkcs12 -export -inkey win-code-signing.key -in win-code-signing.crt -out encrypted-win-code-signing.pfx
+    Enter Export Password: <passphrase>
```
Comment (flotwig) [inline guides/code-signing.md]: since openssl uses "password" here:

```suggestion
    Enter Export Password: <password>
```

**Pull request merged**